src/components/expression/expression-renderer.js

```diff
@@ -968,13 +1118,40 @@ genotet.ExpressionRenderer.prototype.highlightLabelsAfterUpdateData_ =
       this.clickedObject_.column = heatmapData.conditionNames.indexOf(
         this.clickedObject_.conditionName
       );
-      this.signal('expressionClick', this.clickedObject_);
+    }
+    if (this.clickedObject_.row != -1 && this.clickedObject_.column != -1) {
```

Comment:
You are using a special meaning of -1 that indicates undefined row or column. It is not good to have many such constants here and there, unless the constant's meaning is well-known. If you are using indexOf() == -1, that's fine. However, your -1 has special indications and could be better handled by "enum"-like constants instead.

I'm not aware of other files using -1 for non indexOf operations. If they are indeed using, they need to either detailedly comment the usage, or define a constant as well.